src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -109,39 +106,18 @@ public unsafe static SocketError Accept(SafeCloseSocket socketHandle, byte[] soc
 
         private void InnerReleaseHandle()
         {
-            // No-op for Unix.
+            if (_asyncContext != null)
+            {
+                _asyncContext.Close();
```

Comment:
Yes, once non-null always non-null.  Double-closes should be prevented by the SafeHandle infrastructure itself (this runs inside of SafeHandle.ReleaseHandle).